### PR TITLE
[DMA 4/8]: Apply Peripheral pattern

### PR DESCRIPTION
**Pull request by bugadani**

### CODE REVIEW

**Comment by bugadani** [comment]
Don't let me get away with the I8080 change, it's incomplete - asyncness is implied by usage right now, which is bad.

File: hil-test/tests/embassy_interrupt_spi_dma.rs [hunk 1]
```diff
@@ -137,7 +137,7 @@ mod test {
         let other_peripheral = Spi::new_with_config(
             peripherals.SPI3,
             Config {
-                frequency: 100.kHz(),
+                frequency: 10000.kHz(),
```
Comment (bugadani) [inline hil-test/tests/embassy_interrupt_spi_dma.rs:140]: Had to pull this up: https://github.com/esp-rs/esp-hal/pull/2543#discussion_r1843613723

File: esp-hal/src/dma/mod.rs [hunk 2]
```diff
@@ -1585,13 +1593,46 @@ impl RxCircularState {
...
-    type Tx: TxRegisterAccess + InterruptAccess<DmaTxInterrupt>;
+    type Tx: DmaTxChannel;
+
+    /// Sets the priority of the DMA channel.
+    #[cfg(gdma)]
+    fn set_priority(&self, priority: DmaPriority);
+
+    /// Splits the DMA channel into its RX and TX halves.
+    #[cfg(any(esp32c6, esp32h2, esp32s3))] // TODO relax this to allow splitting on all chips
+    fn split(self) -> (Self::Rx, Self::Tx) {
```
Comment (bugadani) [inline esp-hal/src/dma/mod.rs:1622]: Something is missing still, so this isn't useable. The upcoming PRs should fix that.

File: esp-hal/CHANGELOG.md [hunk 3]
```diff
@@ -10,15 +10,22 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- DMA channel objects now implement `Peripheral` (#2526)
+- DMA channel objects are no longer wrapped in `Channel`. The `Channel` drivers are now managed by DMA enabled peripheral drivers. (#2526)
+- The `Dpi` driver and `DpiTransfer` now have a `Mode` type parameter. The driver's asyncness is determined by the asyncness of the `Lcd` used to create it. (#2526)
+
 ### Fixed
 
 ### Removed
 
 - The `configure` and `configure_for_async` DMA channel functions has been removed (#2403)
+- The DMA channel objects no longer have `tx` and `rx` fields. (#2526)
```
Comment (Dominaezzz) [inline esp-hal/CHANGELOG.md:28]: Is this temporary or is the migration guide coming in a later PR?

File: esp-hal/CHANGELOG.md [hunk 4]
```diff
@@ -10,15 +10,22 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 
 - ESP32-S3: Added SDMMC signals (#2556)
-- `dma::{Channel, ChannelRx, ChannelTx}::set_priority` for GDMA devices (#2403)
+- Aded `set_priority` to the `DmaChannel` trait on GDMA devices (#2403, #2526)
```
Comment (Dominaezzz) [inline esp-hal/CHANGELOG.md:13]:
```suggestion
- Added `set_priority` to the `DmaChannel` trait on GDMA devices (#2403, #2526)
```

Comment (Dominaezzz) [reply esp-hal/CHANGELOG.md:13]: (In the future) For completeness should `set_priority` take an rx and tx priority?

**Review by Dominaezzz — APPROVED** [review]
LGTM sans some minor nits

File: esp-hal/CHANGELOG.md (hunk 4 above)
Comment (bugadani) [reply esp-hal/CHANGELOG.md:13]: No, but rx and tx may have set_priority I guess.

**Review by bjoernQ — APPROVED** [review]
Thanks

**Pull request merged**